File: insertionSort.js
const input = [3, 1, 66, 43, 20]

for (const j of input.map((e, i)=>i).slice(1)) {
    let key = input[j]
    let i = j - 1
    while (i > -1 && input[i] > key) {
        input[i + 1] = input[i]
        i = i - 1
    }
    input[i + 1] = key
}

console.log(input);

File: insertion_sort.py
input = [i for i in range(10000000)]
input.reverse()
for i in range(1, len(input)):
    key = input[i]
    j = i - 1
    while j > -1 and input[j] > key:
        input[j + 1] = input[j]
        j = j - 1
    input[j + 1] = key


print(i)
print(input)